hfc/fabric/client.py

```diff
@@ -1936,12 +1940,16 @@ async def query_info(self, requestor, channel_name, peers, decode=True):
         r = []
         for v in res:
             try:
-                if v.response and decode:
-                    chain_info = ledger_pb2.BlockchainInfo()
-                    chain_info.ParseFromString(v.response.payload)
-                    _logger.debug('response status {}'.format(
-                        v.response.status))
-                    return chain_info
+                if v.response:
+                    if decode:
+                        chain_info = ledger_pb2.BlockchainInfo()
+                        chain_info.ParseFromString(v.response.payload)
+                        _logger.debug('response status {}'.format(
+                            v.response.status))
+                        return chain_info
+                    else:
+                        return v.response.payload
+
                 r.append(v)
```

Comment:
@yeasy just wondering about that statements here? 
```
else:
                raise Exception(r)
```
Is there a particular reason to return the response that does need to be decoded wrapped in an exception?

Reply:
@dpdornseifer Hi, David, thanks for the PR. I think line 1952 `raise Exception(r)` here returns an empty list if nothing is here. And it doesn't seem to me there is a particular reason to return it.

Reply:
No special reason. We can return empty list to indicate there's error happend.

Reply:
ok, thx - I kept it like is was